rir/src/ir/BC_inc.h

```diff
@@ -108,6 +109,19 @@ class BC {
         Immediate expected;
         Immediate id;
     };
+    struct AssertTypeArgs {
+        static_assert(
+            sizeof(pir::PirType) == sizeof(Immediate) * 2,
+            "PirType must fit in 2 immediates, or change assert_type_ size");
+        Immediate typeData1;
+        Immediate typeData2;
+        const char* instr;
```

Comment:
please no pointers to the C heap in the bytecode stream! you should know why, when you try to serialize RIR code, this will be a pain.